File: docs/sdks/python/examples/authentication.py
"""Authentication examples"""

import os
from lydian import Lydian, LydianError

# Example 1: API Key Authentication
def api_key_auth():
    lydian = Lydian(api_key=os.environ.get("LYDIAN_API_KEY"))
    cities = lydian.smart_cities.list_cities()
    print(f"Found {len(cities.data)} cities")

# Example 2: OAuth2 Authentication
def oauth2_auth():
    lydian = Lydian(base_url="https://api.lydian.ai/v1")

    # Authenticate with OAuth2
    lydian.authenticate_oauth2(
        client_id=os.environ.get("OAUTH_CLIENT_ID"),
        client_secret=os.environ.get("OAUTH_CLIENT_SECRET")
    )

    # Now make authenticated requests
    cities = lydian.smart_cities.list_cities()
    print(f"Found {len(cities.data)} cities")

# Example 3: Custom configuration
def custom_config():
    lydian = Lydian(
        api_key=os.environ.get("LYDIAN_API_KEY"),
        base_url="https://api.lydian.ai/v1",
        timeout=60,  # 60 seconds
        retry_attempts=5,
        retry_delay=2  # 2 seconds
    )

    cities = lydian.smart_cities.list_cities()
    print(f"Found {len(cities.data)} cities")

# Example 4: Error handling
def error_handling():
    lydian = Lydian(api_key=os.environ.get("LYDIAN_API_KEY"))

    try:
        city = lydian.smart_cities.get_city("non-existent-id")
        print(city)
    except LydianError as e:
        print(f"API Error: {e.message}")
        print(f"Status: {e.status_code}")
        print(f"Code: {e.code}")
        print(f"Details: {e.details}")
    except Exception as e:
        print(f"Unexpected error: {e}")

if __name__ == "__main__":
    print("1. API Key Authentication")
    api_key_auth()

    print("\n2. OAuth2 Authentication")
    oauth2_auth()

    print("\n3. Custom Configuration")
    custom_config()

    print("\n4. Error Handling")
    error_handling()

File: docs/sdks/typescript/examples/authentication.ts
import { Lydian } from '@lydian/sdk';

// Example 1: API Key Authentication
async function apiKeyAuth() {
  const lydian = new Lydian({
    apiKey: process.env.LYDIAN_API_KEY,
  });

  const cities = await lydian.smartCities.listCities();
  console.log('Cities:', cities);
}

// Example 2: OAuth2 Authentication
async function oauth2Auth() {
  const lydian = new Lydian({
    baseUrl: 'https://api.lydian.ai/v1',
  });

  // Authenticate with OAuth2
  await lydian.authenticateOAuth2({
    clientId: process.env.OAUTH_CLIENT_ID!,
    clientSecret: process.env.OAUTH_CLIENT_SECRET!,
  });

  // Now make authenticated requests
  const cities = await lydian.smartCities.listCities();
  console.log('Cities:', cities);
}

// Example 3: Custom configuration
async function customConfig() {
  const lydian = new Lydian({
    apiKey: process.env.LYDIAN_API_KEY,
    baseUrl: 'https://api.lydian.ai/v1',
    timeout: 60000, // 60 seconds
    retryAttempts: 5,
    retryDelay: 2000, // 2 seconds
  });

  const cities = await lydian.smartCities.listCities();
  console.log('Cities:', cities);
}

// Example 4: Error handling
async function errorHandling() {
  const lydian = new Lydian({
    apiKey: process.env.LYDIAN_API_KEY,
  });

  try {
    const city = await lydian.smartCities.getCity('non-existent-id');
    console.log(city);
  } catch (error) {
    if (error instanceof Error) {
      console.error('Error:', error.message);

      // Check if it's a Lydian API error
      if ('statusCode' in error) {
        const apiError = error as any;
        console.error('Status:', apiError.statusCode);
        console.error('Code:', apiError.code);
        console.error('Details:', apiError.details);
      }
    }
  }
}

// Run examples
async function main() {
  console.log('1. API Key Authentication');
  await apiKeyAuth();

  console.log('\n2. OAuth2 Authentication');
  await oauth2Auth();

  console.log('\n3. Custom Configuration');
  await customConfig();

  console.log('\n4. Error Handling');
  await errorHandling();
}

main().catch(console.error);
